test/ejabberd_tests/tests/ejabberdctl_SUITE.erl

```diff
@@ -595,10 +608,175 @@ stats_host(Config) ->
+        OldTimestamp = fallback_timestamp(10, now()),
+        OfflineOld = generate_offline_message(JidRecordAlice, JidRecordBob, Msg1, OldTimestamp),
+        OfflineNew = generate_offline_message(JidRecordAlice, JidRecordBob, Msg2, now()),
+        {jid, _, _, _, LUser, LServer, _} = JidRecordBob,
+        rpc_call(mod_offline_backend, write_messages, [LUser, LServer, [OfflineOld, OfflineNew], 100]),
+        %% when
+        {_, _} = ejabberdctl("delete_old_messages", ["1"], Config),
+        {ok, SecondList} = rpc_call(mod_offline_backend, pop_messages, [LUser, LServer]),
+        %% then
+        1 = length(SecondList)
+    end).
+
+remove_expired_messages_test(Config) ->
+    escalus:story(Config, [{mike, 1}], fun(_) ->
+        %% given
+        JidA = nick_to_jid(mike, Config),
+        JidB = nick_to_jid(kate, Config),
+        JidRecordMike = rpc_call(jlib, binary_to_jid, [JidA]),
+        JidRecordKate = rpc_call(jlib, binary_to_jid, [JidB]),
+        Msg1 = escalus_stanza:chat_to(<<"kate@localhost">>, "Rolling stones"),
+        Msg2 = escalus_stanza:chat_to(<<"kate@localhost">>, "Arctic monkeys!"),
+        Msg3 = escalus_stanza:chat_to(<<"kate@localhost">>, "More wine..."),
+        Msg4 = escalus_stanza:chat_to(<<"kate@localhost">>, "kings of leon"),
+        OldTimestamp = fallback_timestamp(10, now()),
+        ExpirationTime = fallback_timestamp(2, now()),
+        ExpirationTimeFuture= fallback_timestamp(-5, now()),
+        OfflineOld = generate_offline_expired_message(JidRecordMike, JidRecordKate, Msg1, OldTimestamp, ExpirationTime),
+        OfflineNow = generate_offline_expired_message(JidRecordMike, JidRecordKate, Msg2, now(), ExpirationTime),
+        OfflineFuture = generate_offline_expired_message(JidRecordMike, JidRecordKate, Msg3, now(), ExpirationTimeFuture),
+        OfflineFuture2 = generate_offline_expired_message(JidRecordMike, JidRecordKate, Msg4, OldTimestamp, ExpirationTimeFuture),
+        {jid, _, _, _, LUser, LServer, _} = JidRecordKate,
+        rpc_call(mod_offline_backend, write_messages, [LUser, LServer, [OfflineOld, OfflineNow, OfflineFuture, OfflineFuture2], 100]),
+        %% when
+        {_, _} = ejabberdctl("delete_expired_messages", [], Config),
+        {ok, SecondList} = rpc_call(mod_offline_backend, pop_messages, [LUser, LServer]),
+        %% then
+        2 = length(SecondList)
+    end).
+
 %%-----------------------------------------------------------------
 %% Helpers
 %%-----------------------------------------------------------------
 
+
+nick_to_jid(UserName, Config) when is_atom(UserName) ->
+    UserSpec = escalus_users:get_userspec(Config, UserName),
+    escalus_utils:jid_to_lower(escalus_users:get_jid(Config, UserSpec)).
+
+generate_offline_message(From, To, Msg, TimeStamp) ->
+    {jid, _, _, _, LUser, LServer, _} = To,
+    #offline_msg{us={LUser, LServer}, timestamp=TimeStamp, expire=never,from=From, to=To, packet=Msg}.
```

Comment:
Putting space before and after `=` will make the code more readable. If the line gets too long you can always break it into two or more lines.